Shared/Samples/Apply scheduled updates to preplanned map area/ApplyScheduledUpdatesToPreplannedMapAreaView.swift

```diff
@@ -0,0 +1,200 @@
+import SwiftUI
+
+struct ApplyScheduledUpdatesToPreplannedMapAreaView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// A Boolean value indicating whether the alert for
+    /// scheduled updates is presented.
+    @State private var updatesAlertIsPresented = false
+    
+    /// A Boolean value indicating whether the alert for
+    /// no available updates is presented.
+    @State private var noUpdatesAlertIsPresented = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    var body: some View {
+        MapView(map: model.map)
+            .errorAlert(presentingError: $error)
+            .task {
+                do {
+                    try await model.setUp()
+                } catch {
+                    self.error = error
+                }
+            }
+            .onChange(of: model.updatesInfo == nil) { _ in
+                if let updatesInfo = model.updatesInfo {
+                    // Handle the updates info from the offline map sync task.
+                    handleUpdatesInfo(updatesInfo)
+                }
+            }
+            .overlay {
+                // Show a progress view for the offline map sync job that downloads
+                // the available updates.
+                if let progress = model.offlineMapSyncJob?.progress {
+                    VStack(spacing: 16) {
+                        ProgressView(progress)
+                            .progressViewStyle(.linear)
+                            .frame(maxWidth: 200)
+                    }
+                    .padding()
+                    .background(.regularMaterial)
+                    .clipShape(RoundedRectangle(cornerRadius: 15))
+                    .shadow(radius: 3)
+                }
+            }
+            .alert("Scheduled Updates Available", isPresented: $updatesAlertIsPresented) {
+                Button("Cancel", role: .cancel, action: {})
+                Button("Apply") {
+                    Task {
+                        do {
+                            try await model.applyScheduledUpdates()
+                        } catch {
+                            self.error = error
+                        }
+                    }
+                }
+            } message: {
+                // Get the download size for the update.
+                let downloadSizeString = ByteCountFormatter.string(
+                    from: Measurement(
+                        value: Double(model.updatesInfo?.scheduledUpdatesDownloadSize ?? .zero),
+                        unit: UnitInformationStorage.bytes
+                    ),
+                    countStyle: .file
+                )
+                Text("A \(downloadSizeString) update is available. Would you like to apply it?")
+            }
+            .alert("Scheduled Updates Unavailable", isPresented: $noUpdatesAlertIsPresented) {
+                Button("OK", action: {})
```

Comment:
`alert` shows an "OK" button by default, so I am not sure this button is needed.